api/src/modules/tasks/dto/update-task-with-controller.dto.ts

```diff
@@ -0,0 +1,23 @@
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { IsNotEmpty, IsOptional, IsString, IsUUID } from 'class-validator';
+import { TASK_STATUS } from 'modules/tasks/task.entity';
+
+export class UpdateTaskWithControllerDto {
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsUUID()
+  taskId: string;
+
+  @ApiProperty()
+  @IsNotEmpty()
+  @IsString()
+  newStatus: TASK_STATUS;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  newData?: Record<string, any>;
+
+  @ApiPropertyOptional()
+  @IsOptional()
+  newErrors?: Error;
```

Comment:
I suspect the type here is wrong

Reply:
Actually the method taking care of updating a task's status if any error is thrown, consumes an error to then push an object with it's name as key and the message as value.

But this happens within the API, so I think that typing a new error as Error when it is meant to be updated via controller makes sense.

As a note I'd like to add that I think we should not allow to update an error on a task via controller as any possible error would happen withing the API.
Thoughts on this?

Reply:
I agree with that last thought (and pretty much all of the rest).